File: app/src/main/java/com/example/xeltim/biologyjourney/MainActivity.java
package com.example.xeltim.biologyjourney;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.support.design.widget.FloatingActionButton;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends Activity {

    Button button;
    FloatingActionButton fab;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.activity_main);

        button = (Button)findViewById(R.id.button2);
        fab = (FloatingActionButton) findViewById(R.id.floatingActionButton);

        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Settings();
            }
        });

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                isNetworkConnected();
                boolean Status = isNetworkConnected();
                if(Status == true)
                Start();
                else
                    Toast.makeText(MainActivity.this, "No Internet Connection!", Toast.LENGTH_SHORT).show();
            }
        });
    }

    protected void Start(){
        Intent intent = new Intent(this, Home.class);
        startActivity(intent);
    }

    protected void Settings(){
        Intent intent = new Intent(this, SettingsActivity.class);
        startActivity(intent);
    }

    private boolean isNetworkConnected() {
        ConnectivityManager cm = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);

        return cm.getActiveNetworkInfo() != null;
    }

}

File: app/src/main/java/com/example/xeltim/biologyjourney/MainActivity.kt
package com.example.xeltim.biologyjourney

import android.app.Activity
import android.app.AlarmManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.view.Window
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : Activity(){

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button2.setOnClickListener(){
            isNetworkConnected()
            var Status : Boolean = isNetworkConnected()
            if(Status === true){
                Start()
                startAlarm()
            }else{
                Toast.makeText(this@MainActivity, "No Internet Connection!", Toast.LENGTH_SHORT).show()
            }
        }

    }

    protected fun Start():Unit{
        var intent : Intent = Intent(this, Home::class.java)
        startActivity(intent)
    }

    protected fun Settings() : Unit{
        var intent : Intent = Intent(this, SettingsActivity::class.java)
        startActivity(intent);
    }

    protected fun isNetworkConnected() : Boolean{
        var cm : ConnectivityManager = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return cm.getActiveNetworkInfo() != null
    }

    private fun startAlarm(){
        var calendar:Calendar = Calendar.getInstance()

        calendar.set(Calendar.HOUR_OF_DAY, 9)
        calendar.set(Calendar.MINUTE, 45)
        calendar.set(Calendar.SECOND, 0)

        val pendingIntent = PendingIntent.getBroadcast(this, 100,
                Intent(this, AlarmReciver::class.java), PendingIntent.FLAG_UPDATE_CURRENT)

        val alarmManager = this.getSystemService(Context.ALARM_SERVICE) as AlarmManager

        alarmManager.setRepeating(AlarmManager.RTC_WAKEUP, calendar.timeInMillis,
                AlarmManager.INTERVAL_FIFTEEN_MINUTES, pendingIntent)
    }

}